### ISSUE TITLE
Build Error：No rule to make target VBoxBldProg-libssl.lib, needed by bldRTSignTool.exe

### ISSUE THREAD

**Issue by guojia1992** [VirtualBoBs/build-virtualbox-in-windows#8]
![image](https://user-images.githubusercontent.com/31852817/209748948-e2f325d7-eea8-4898-a6f0-2be4f1d62db7.png)

I am trying compile vbox and everything is ok before execute build.py.
The error info is showing as the image, could any one help?

**Comment by guojia1992** [comment]
More info:  I use vbox-6.1.36  source code

**Comment by jidoc01** [comment]
I will inspect this case soon. Thank you for reporting!

**Comment by arminanton** [comment]
I'm having the same issue. Were you able to build version 7.0.4 on Windows 11?

**Comment by guojia1992** [comment]
> I'm having the same issue. Were you able to build version 7.0.4 on Windows 11?

I am building version 6.1.36 on Windows 11 and failed as the issue above. not trying 7.0.4 yet

**Comment by guojia1992** [comment]
> I will inspect this case soon. Thank you for reporting!

My build is on Windows11  and source code version 6.1.36。
Please let me know if there is any progress , thank you very much。

**Comment by xiaonaiquan** [comment]
> 我遇到了同样的问题。您是否能够在 Windows 11 上构建 7.0.4 版？

Are you making any progress on building Virtual box 7.0.4?

**Comment by xiaonaiquan** [comment]
> I'm having the same issue. Were you able to build version 7.0.4 on Windows 11?

Are you making any progress on building Virtual box 7.0.4?

**Comment by xiaonaiquan** [comment]
@guojia1992  Hi, Are you making progress with the compilation process?

**Comment by xiaonaiquan** [comment]
I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk

**Comment by arminanton** [comment]
> I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk

That's awesome!!

Could you try building version 7.0.4 x64 on Windows 11? Also, build the with the guest additions tools as well please.

I heard that only version 7+ can fully function on windows 11...

**Comment by xiaonaiquan** [comment]
> > I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk
> 
> That's awesome!!
> 
> Could you try building version 7.0.4 x64 on Windows 11? Also, build the with the guest additions tools as well please.
> 
> I heard that only version 7+ can fully function on windows 11...

I'm sorry, I don't have a windows11 system computer, you can try to compile, if there is a problem, you can leave a message here, I will reply you when I see it.

**Comment by guojia1992** [comment]
> I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk

COOL！

**Comment by guojia1992** [comment]
> I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk

COOL！

**Comment by guojia1992** [comment]
> I have solved this problem and need to add "VBOX_WITH_BLD_RTSIGNTOOL_SIGNING :=" to AutoConfig.kmk

COOL! I will try this way later

**Comment by guojia1992** [comment]
Hello ALL， share my updats：

1、I have successfully build vbox by replacing the source code with version 6.1.30 on Windows 11.  Version 6.1.30 don‘t  have the issue “No rule to make target VBoxBldProg-libssl.lib”.  If someone else encounters the problem “make356-1.sh: Syntax error: "(" unexpected”,  use 'mklink' command to make a hard link to replace VS2010 install path with NO SPACES, it  should work.  
2、I have also successfully build Guest Addition by some other extra operations:
   a. add VBOX_ONLY_ADDITIONS := 1 to LocalConfig.kmk
   b. Install vs2013  with update 5( below update 4 have compile errors).
   c. Install wddk 8.1
   d. set PATH_SDK_WINDDK80  := /install-path-to-wddk81 in the file  kBuild/sdks/WINDDK80.kmk
![image](https://user-images.githubusercontent.com/31852817/213603195-91aa8ec5-60d2-4409-a631-4467c14e97a4.png)
   e.  set PATH_TOOL_VCC120AMD64 := /install-path-to-vs2013/VC  in the file  kBuild/tools/VCC120AMD64.kmk
![image](https://user-images.githubusercontent.com/31852817/213604224-030d8cc6-639c-476b-abf9-e8c1705b2df7.png)

PS： There must be a more elegant way to set this two environment variables. :)

For d and e ,  use 'mklink' command to make a hard link if install path have any spaces.

Good Luck!

**Comment by jidoc01** [comment]
@xiaonaiquan @guojia1992 I am really thankful for your sharing experiences! And I am sorry that I am currently not able to test the issues, but I will check them ASAP.

**Comment by Whibelf2** [comment]
Hello guys, are yours able compile virtualbox 7 ?

**Comment by Whibelf2** [comment]
> @xiaonaiquan @guojia1992 I am really thankful for your sharing experiences! And I am sorry that I am currently not able to test the issues, but I will check them ASAP.

You have plan to update this script to build Virtualbox 7.x.x ?

**Comment by xiaonaiquan** [comment]
> > @xiaonaiquan @guojia1992 I am really thankful for your sharing experiences! And I am sorry that I am currently not able to test the issues, but I will check them ASAP.
> 
> You have plan to update this script to build Virtualbox 7.x.x ?

I'm trying, but there's something wrong with compiling.

**Comment by arminanton** [comment]
Any news on the Virtualbox 7.x.x?

**Comment by arminanton** [comment]
Seems like someone was able to successfully build it on 7.x.x as well:
https://github.com/bruce30262/build-virtualbox-in-windows

**Comment by xiaonaiquan** [comment]
> 

No, I tried many times, but failed, virtualbox code error occurred
